### PR TITLE
Server timeout returns an incorrect error

### PR DESCRIPTION
**Pull request by smarterclayton**
Not a valid Status object in JSON

Part of #42163

### CODE REVIEW

**Comment by smarterclayton** [comment]
Incorrect error output

@kubernetes/sig-api-machinery-pr-reviews

**Comment by smarterclayton** [comment]
@k8s-bot kops aws e2e test this

File: staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go [hunk 1]
```diff
@@ -67,14 +70,18 @@ func TestTimeout(t *testing.T) {
...
 	}
 	if res.StatusCode != http.StatusGatewayTimeout {
 		t.Errorf("got res.StatusCode %d; expected %d", res.StatusCode, http.StatusServiceUnavailable)
 	}
 	body, _ = ioutil.ReadAll(res.Body)
-	if !strings.Contains(string(body), timeoutErr.Error()) {
-		t.Errorf("got body %q; expected it to contain %q", string(body), timeoutErr.Error())
+	status := &metav1.Status{}
+	if err := json.Unmarshal(body, status); err != nil {
+		t.Fatal(err)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go]: this leaks the handle waiting on sendResponse

Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go]: only in error case. so it does not matter.

**Comment by smarterclayton** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**